presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -334,6 +338,49 @@ public ConnectorPageSource getPageSource(
         }
     }
 
+    Optional<BitSet> getRowsFromUuid(FileSystem fileSystem, Optional<UUID> deltaShardUuid)
+    {
+        if (!deltaShardUuid.isPresent()) {
+            return Optional.empty();
+        }
+        try (OrcDataSource dataSource = openShard(fileSystem, deltaShardUuid.get(), defaultReaderAttributes)) {
+            AggregatedMemoryContext systemMemoryUsage = newSimpleAggregatedMemoryContext();
+            OrcReader reader = new OrcReader(
+                    dataSource,
+                    ORC,
+                    defaultReaderAttributes.getMaxMergeDistance(),
+                    defaultReaderAttributes.getTinyStripeThreshold(),
+                    HUGE_MAX_READ_BLOCK_SIZE,
+                    orcFileTailSource,
+                    new StorageStripeMetadataSource());
+            if (reader.getFooter().getNumberOfRows() >= Integer.MAX_VALUE) {
+                throw new IOException("File has too many rows");
+            }
+            ImmutableMap.Builder<Integer, Type> includedColumns = ImmutableMap.builder();
+            includedColumns.put(0, BIGINT);
+            OrcBatchRecordReader recordReader = reader.createBatchRecordReader(
+                    includedColumns.build(),
+                    OrcPredicate.TRUE,
+                    DEFAULT_STORAGE_TIMEZONE,
+                    systemMemoryUsage,
+                    INITIAL_BATCH_SIZE);
+            BitSet bitSet = new BitSet();
+            while (recordReader.nextBatch() > 0) {
+                Block block = recordReader.readBlock(BIGINT, 0);
```

Comment:
Let's use `BOOLEAN` to take advantage of RLE. `BIGINT` reads are in general more expensive.